stock_release_channel/models/stock_release_channel.py

```diff
@@ -873,15 +884,22 @@ def action_unlock(self):
         self.write({"state": "open"})
 
     def action_sleep(self):
+        # TODO: Add safe argument in function arguments in further version to
+        # avoid context call
+        safe_sleep = self.env.context.get("release_channel_safe_sleep", False)
```

Comment:
Why didn't you do it now. 
Couldn't you move the code of `action_sleep` into a new method for example `def _put_to_sleep(safe=False)` and call this then from `action_sleep` and `action_safe_sleep`?

Reply:
Ok, yes, good point